presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueryFramework.java

```diff
@@ -141,12 +141,17 @@ protected void assertQuery(Session session, @Language("SQL") String actual, @Lan
         QueryAssertions.assertQuery(queryRunner, session, actual, h2QueryRunner, expected, false, false);
     }
 
-    protected void assertQuery(Session session, @Language("SQL") String actual, @Language("SQL") String expected, Consumer<Plan> planAssertion)
+        protected void assertQuery(Session session, @Language("SQL") String actual, @Language("SQL") String expected, Consumer<Plan> planAssertion)
     {
         checkArgument(queryRunner instanceof DistributedQueryRunner, "pattern assertion is only supported for DistributedQueryRunner");
         QueryAssertions.assertQuery(queryRunner, session, actual, h2QueryRunner, expected, false, false, planAssertion);
     }
 
+    protected void assertQuery(Session session, @Language("SQL") String actual, Session referenceSession, Optional<String> expected)
```

Comment:
Let's not add methods with Optional<String> query parameters. Elsewhere, the necessary functionality is achieved by having two methods: with and without query parameter.

```
    protected void assertQuery(Session session, @Language("SQL") String actual, Session expectedSession)
    {
        QueryAssertions.assertQuery(queryRunner, session, actual, expectedSession, actual);
    }

    protected void assertQuery(Session session, @Language("SQL") String actual, Session expectedSession, @Language("SQL") String expected)
    {
        QueryAssertions.assertQuery(queryRunner, session, actual, expectedSession, expected);
    }
```